### PR TITLE
all color themes: treat comment docstrings as comments too

### LINKED ISSUE #182163: Python multi-line comment strings should be treated as comments in all themes
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.78.1
- OS Version: Linux Ubuntu 20.04

Steps to Reproduce:

1. Set the color theme to Monokai. Comment block strings should be highlighted the same as single-line comments, just like Monokai in Sublime Text, but in VSCode they are not. 

Fixed by PR #182162 <-- **see screenshots here** too!

### PR DESCRIPTION
**Pull request by ElectricRCAircraftGuy**
This treats Python comment block docstrings as comments too, so they are subdued, like in Sublime Text, so we can focus on the code. 

Before this change (bad, because the Python block comment strings are an obnoxious yellow): 
image

After (good, because now they are treated like comments, and subdued):
image

Notice that regular multi-line block strings (`str` in the code above) still show up in yellow, as they should.

I wrote about this here: [How to subdue Python block comment strings in VSCode in order to look like Sublime Text's Monokai](https://stackoverflow.com/a/76224318/4561887)

Fixes #182163

### CODE REVIEW

**Comment by ElectricRCAircraftGuy** [comment]
I have tested this PR by making and running this custom theme extension: https://github.com/ElectricRCAircraftGuy/GS_vs_code_custom_theme

**Comment by aeschli** [comment]
It makes sense to render docstrings as comments. Would you mind adding that rule to the other built-in themes as well?

**Comment by ElectricRCAircraftGuy** [comment]
> [aeschli](/aeschli)

Done!

**Comment by ElectricRCAircraftGuy** [comment]
> It makes sense to render docstrings as comments. Would you mind adding that rule to the other built-in themes as well?

@aeschli , done!

**Comment by aeschli** [comment]
Thanks! 
I would suggest to include all doc strings, including the single line and raw doc strings.
The rule would be `string.quoted.docstring`, not limited to python, assuming that other languages using that scope use it for documentation as well (I didn't find other grammars using it)

**Comment by ElectricRCAircraftGuy** [comment]
> Thanks! I would suggest to include all doc strings, including the single line and raw doc strings. The rule would be `string.quoted.docstring`, not limited to python, assuming that other languages using that scope use it for documentation as well (I didn't find other grammars using it)

@aeschli , done.

**Review by aeschli — APPROVED** [review]

**Review by dbaeumer — APPROVED** [review]

**Comment by k98kurz** [comment]
How do I reverse this locally? I preferred it the way it was. Sudden an unexplained changes after a restart are far more obnoxious than the orange color -- at least it looked good, which I can't say for this ugly dark green.

**Comment by starball5** [comment]
Related on Stack Overflow: [Restarted Visual Studio Code, and now Python docstrings are an ugly dark green. How do I return it to the original orange?](https://stackoverflow.com/q/76436694/11107541)

**Comment by torext** [comment]
Very much agreed that sudden and unexplained changes like this, which ultimately boil down to purely someone's personal preference, are quite obnoxious. While the change makes conceptually sense, it would be better in future to maintain current behaviour while enacting the change.

**Comment by ElectricRCAircraftGuy** [comment]
> Very much agreed that sudden and unexplained changes like this, which ultimately boil down to purely someone's personal preference, are quite obnoxious. While the change makes conceptually sense, it would be better in future to maintain current behaviour while enacting the change.

I'm not with VSCode, but I'm wondering: how does one maintain current behavior while enacting the change? They seem so me to be mutually exclusive events. 

Part of the pain is that the syntax highlighting schemes may need to be updated once the change happens, too, but it's not practical to expect that to happen at the same time, nor does it make sense to update all schemes to undo the very behavior this change makes. I think it's just an unfortunate side effect of having wrong behavior that some people will prefer the wrong behavior to continue because they got used to it.

**Comment by ElectricRCAircraftGuy** [comment]
For anyone wishing to customize your theme to go back to how it was for your syntax highlighter, read my answer here to learn how to do that. It contains info about finding the names and colors of sections you like and how to change the ones you don't:

[How to subdue Python block comment strings in VSCode in order to look like Sublime Text's Monokai](https://stackoverflow.com/a/76224318/4561887)

**Comment by lkct** [comment]
> It makes sense to render docstrings as comments.

> I think it's just an unfortunate side effect of having wrong behavior that some people will prefer the wrong behavior to continue because they got used to it.

@ElectricRCAircraftGuy @aeschli 
Sorry, but I even didn't get the point from the very beginning: why should we treat docstrings as comments?

I use docstrings as ***doc***strings, containing information, e.g., for functions/methods, the summary of functionality and signature. These should be more important than just a "comment" and be highlighted differently.
Why is this "wrong behavior"?

**Comment by k98kurz** [comment]
> I use docstrings as _**doc**_strings, containing information, e.g., for functions/methods, the summary of functionality and signature. These should be more important than just a "comment" and be highlighted differently. Why is this "wrong behavior"?

Docstrings are essentially block comments at the top of a scope, so I can see why it would seem appropriate to treat them as block comments. However, they are semantically different from a regular block comment, so the appropriateness of treating them the same is only a seeming. I am unaware of any other language having an equivalent to docstrings -- IDEs can parse certain block comments as docstrings, but `__doc__` is a part of the Python language and not a part of, say, EcmaScript, so it should this language feature should be treated differently since it is explicitly a different language feature.

@ElectricRCAircraftGuy I think you replaced a wrong behavior with a different wrong behavior. Unless there is a way to scope theme colors to specific languages or to semantically differentiate Python docstrings from simulated docstrings in other languages, VS Code seems to limit us to choosing between two wrong behaviors.

**Comment by lkct** [comment]
> I am unaware of any other language having an equivalent to docstrings

That's what I've missed! Indeed in other languages like C++, we put documentation in comments.
And YES, you're right. In Python `__doc__` docstrings are part of the language feature, and I agree with differentiation.

> VS Code seems to limit us to choosing between two wrong behaviors.

From the SO answer references above, there's a `string.quoted.docstring.multi.python`, but I'm not sure how we should use it to differentiate Python docstrings.

**Comment by yomajo** [comment]
Wait, so this got implemented, because dude was using fucked up theme?

**Comment by k98kurz** [comment]
> Wait, so this got implemented, because dude was using fucked up theme?

Not exactly. but kinda. All themes had a wrong behavior regarding multiline comments in Python, and now all themes have a wrong behavior regarding docstrings in Python.

**Comment by lkct** [comment]
> Wait, so this got implemented, because dude was using fucked up theme?

However, the problem is no one commented with an objection before it was merged.

**Comment by lobsterkatie** [comment]
FWIW, I saw the title of this PR in the release notes and thought "Finally!" I am very much +1 on this change.

I understand that semantically, docstrings and comments aren't the same thing, but IMHO they are used for enough of the same purpose (documentation ignored by the interpreter*) that they should be lumped together, and both be dimmed, since they don't affect the behavior of the function/method/module/whatever. Conversely, docstrings, while they are not comments, are also not strings in the normal sense, and therefore should not be colored the same way. 

*I'm leaving aside the issue of doctests, which of course the interpreter does handle, but AFAIK they're not treated differently than any other part of a docstring, so for the purposes of this conversation I'm lumping them in with non-runnable docstring content.

**Comment by k98kurz** [comment]
> I understand that semantically, docstrings and comments aren't the same thing, but IMHO they are used for enough of the same purpose (documentation ignored by the interpreter*) that they should be lumped together, and should both be dimmed, since they don't affect the behavior of the function/method/module/whatever. Conversely, docstrings, while they are not comments, are also not strings in the normal sense, and therefore should not be colored the same way.

The interpreter does not actually ignore them in Python -- it puts them on a special `__doc__` attribute to be used by the built-in `help` function. Perhaps there is an argument that they should be handled as their own special case, but which wrong behavior (i.e. whether they should be incorrectly handled as strings or incorrectly handled as block comment) is a matter of personal preference.

**Comment by lobsterkatie** [comment]
> The interpreter does not actually ignore them in Python -- it puts them on a special __doc__ attribute to be used by the built-in help function.

Okay, yes, fair enough, but for the general running of whatever code I'm writing, they function much more like comments than like values, i.e., the 99% case is that my code doesn't involve `__doc__` or `help`, and therefore I can safely ignore them when reading through code and trying to figure out what it does.

**Comment by lkct** [comment]
> and I can safely ignore them when reading through code and trying to figure out what it does.

Seems we have different opinions on docstrings. I think docstrings exist to help us figure out what it does without reading through code (especially when we just want to use a function/class etc. and don't care about the internals).

**Comment by Afrozodiacc** [comment]
> > I understand that semantically, docstrings and comments aren't the same thing, but IMHO they are used for enough of the same purpose (documentation ignored by the interpreter*) that they should be lumped together, and should both be dimmed, since they don't affect the behavior of the function/method/module/whatever. Conversely, docstrings, while they are not comments, are also not strings in the normal sense, and therefore should not be colored the same way.
> 
> The interpreter does not actually ignore them in Python -- it puts them on a special `__doc__` attribute to be used by the built-in `help` function. Perhaps there is an argument that they should be handled as their own special case, but which wrong behavior (i.e. whether they should be incorrectly handled as strings or incorrectly handled as block comment) is a matter of personal preference.

Agreed. docstrings are documentation and not comments. There are python projects such as `sphinx` which automatically generate documents using docstrings. docstrings are not equivelant to comments hence they should not be coloured equivelantly. An option to collapse docstrings in VSCode much like you can collapse classes or methods may be more suitable than colouring them the same as comments for those which want to "declutter" the code space. VSCode also has special treatment of python docstrings already:

image

If a project is using docstrings as a place for comments, then I'd suggest the developers use comments for comments and docstrings for documentation.

**Comment by alexsaurber** [comment]
This PR was not the way to go about "fixing" this.

If the convention for docstrings was to treat them like comments and not documentation that is actually functionally accessible text, then we would use comment notation. Even PEP8 recommends multiline comments to begin with `#` on every line. These should have been read before changing the highlighting convention of millions of users to a new incorrect format. [PEP8: Block Comments](https://peps.python.org/pep-0008/#block-comments). [PEP257: Docstring Conventions](https://peps.python.org/pep-0257/). Strings and docstrings have the same syntactic highlighting in IDLE. Call me silly but that seems pretty intentional. Additionally, they're called Doc*strings*, not doc*comments*. Why correct an error with an even worse error?

**Comment by lobsterkatie** [comment]
I feel like we're missing the forest for the trees here. 

We all agree that string literals, docstrings, and comments are each their own thing, with different (albeit often overlapping) uses, syntax, integration with tooling, etc.

One option is to make all three different tokens (and therefore potentially all three different colors), as proposed in https://github.com/microsoft/vscode/issues/184836.  Another option is to lump docstrings in with either string literals (as has been the case up 'til now) or comments (as is the case after this change).

The implicit point of this PR, which I'm also arguing for, is that in the end, docstrings are _documentation_. They're English sentences. They're explanations. Changing them doesn't change how a function works, or the result it gives. This makes them fundamentally different than literal values (string or otherwise), but very much akin to comments. (This is why in JS they're grey, just like comments, and always have been.)

Bottom line, in the same way that I want to be able to tune out comments, I want to be able to tune out docstrings, too. I want them both to be there, each serving their purpose, available if I want them, but neither one grabbing my attention. And the most straightforward way to do that is to dim them both, which is what this PR does.

**Comment by PeterJCLaw** [comment]
> However, the problem is no one commented with an objection before it was merged.

Kinda? This PR was open for four days before it was merged (May 11th until 15th). That didn't leave a lot of time for people to comment on a rather substantial change to the UX. I also can't see any attempt to gather input from the Python community, either in general or even the developers of the vscode-python or related extensions. The issue which the PR was marked as fixing was opened _after_ the PR was, implying that no prior requests for this change existed (a quick skim of the issues on this repo seems to agree).

An issue (https://github.com/microsoft/pylance-release/issues/4429) was raised on the 22nd May reporting this as a bug in the insiders build of VSCode, though unfortunately it was combined with a more general syntax highlighting issue and misattributed to pylance rather than VSCode itself. The discussion there suggests that even internal Microsoft folks were unaware of this change!

> I'm assuming above that docstrings were separately configurable before this PR.

Looking at the diff of this PR and the workarounds which people are applying to fix this issue, customising this per-user seems to have already been possible. This change only flips the default so that VSCode itself now treats some Python strings as comments.

---

A conflating factor here is that the parser seems to be imprecise in what it considers docstrings, at least relative to what the interpreter will actually attach to a `__doc__`:
``` python
def foo():
    "docstring" # ends up in `foo.__doc__`, now considered a comment
    print(2)
    "not a docstring" # but still now considered a comment
    return
```

---

Personally I would ask the VSCode team to:
* revert this ASAP, given the disruption it's caused
* open a wider discussion about what the ideal solution is here, involving the vscode-python team and the wider community
* in the meanwhile, publish information about how users can customise this
* maybe reflect on how such an impactful change was made with minimal input

**Comment by lkct** [comment]
> But PR's like this one **should not** change the look of a user's IDE on update and **ideally should** implement "conceptually correct" behaviour once and for all.

Agreed. That's the point I've been making: we should make things "conceptually correct"; if we can't (for the moment), we should choose the way that minimizes UX impact.

> P.S.: I'm assuming above that docstrings were separately configurable before this PR.

Yes, they are, just like the workaround (see SO links above). However, all "literal string statements" currently share the token for "docstrings", and we can only render them as either docstrings or comments.
Note: docstrings do not necessarily look like strings (in an expression); it's just a choice by the themes. As @alexsaurber pointed out, "Strings and docstrings have the same syntactic highlighting in IDLE."

---

> Kinda? This PR was open for four days before it was merged (May 11th until 15th). That didn't leave a lot of time for people to comment on a rather substantial change to the UX.

I don't know how many people are the same, but for me, I actually only looked into this repo after running into the current problem. However, a longer time may still help more people come across this PR.

> I also can't see any attempt to gather input from the Python community, either in general or even the developers of the vscode-python or related extensions.

This is indeed a problem. But it could be because this is treated as a bugfix instead of a feature change.

> The issue which the PR was marked as fixing was opened _after_ the PR was, implying that no prior requests for this change existed

Actually, I sometimes do the similar thing if there's a quick bugfix. I think the core problem here is whether this is a bugfix (which only improves things) or a feature change (which may have a negative impact, at least on part of the users). It's better to gather community inputs before a feature change, while bugfixes may be merged through a faster track. Around this issue, the author thinks this is a bugfix, but I prefer to take it as a feature change. However, the maintainers should rectify how to categorize.

**Comment by lkct** [comment]
> A conflating factor here is that the parser seems to be imprecise in what it considers docstrings, at least relative to what the interpreter will actually attach to a `__doc__`:

I have summarized the behavior in #184836, Python itself has definitions, and the python language server (pylance) has its own interpretation. Therefore I'm proposing to add new tokens and let the language server decide how they're assigned to every string.

For your convenience, I paste the relevant information here:

[Python Docs](https://docs.python.org/3/glossary.html#term-docstring):
> docstring
> A string literal which appears as the first expression in a class, function or module. While ignored when the suite is executed, it is recognized by the compiler and put into the __doc__ attribute of the enclosing class, function or module. Since it is available via introspection, it is the canonical place for documentation of the object.

That is, only the first statement in a scope can be a "docstring".

Python extension of vscode:
```python
class A:
    """This is the docstring for A. Shown on hover."""

    B = 1
    """Shown on hover for B."""

    def c(self, x:int)->int:
        """This is the docstring for c. Shown on hover."""
        d = 1
        """This is nothing but a comment."""

e = 1
"""Shown on hover for e."""
```
For `B` and `e` the strings are not docstrings, but pylance treats them "docstring-like".

---

<details>
<summary>Slightly off-topic</summary>

> The discussion there suggests that even internal Microsoft folks were unaware of this change!

MS is quite large. I think it's normal that the left hand is unaware of what the right hand is doing.
</details>

**Comment by lobsterkatie** [comment]
> > I feel like we're missing the forest for the trees here.
> 
> I feel like you're just reiterating your personal preference?
>
> ...
>
> Docstrings are not comments, the former are programmatically accessible, the latter aren't, and that is that.

Here's what I meant:

Trees: Docstrings and comments are not the same thing, for x, y, z technical reasons. (I think we all agree here.)
Forest: Docstrings are documentation. Comments are also documentation. (I actually think we agree here, too.)

Other than the questions around process (which are valid, but separate), the only part that's opinion/preference is which one of those two is most important. And on that topic I feel like we've all expressed our opinions here both thoroughly and cogently, so at this point I will leave this in the hands of the maintainers.

**Comment by Afrozodiacc** [comment]
> > > I feel like we're missing the forest for the trees here.
> > 
> > 
> > I feel like you're just reiterating your personal preference?
> > ...
> > Docstrings are not comments, the former are programmatically accessible, the latter aren't, and that is that.
> 
> Here's what I meant:
> 
> Trees: Docstrings and comments are not the same thing, for x, y, z technical reasons. (I think we all agree here.) Forest: Docstrings are documentation. Comments are also documentation. (I actually think we agree here, too.)
> 
> Other than the questions around process (which are valid, but separate), the only part that's opinion/preference is which one of those two is most important. And on that topic I feel like we've all expressed our opinions here both thoroughly and cogently, so at this point I will leave this in the hands of the maintainers.

I'd swap the forest and trees in your example. Ultimately the editor needs to respect the python language and the facts are that:
- strings, string literal and docstrings are objects and that docstrings are special attributes of other objects
- comments are not python objects

The fact that docstrings are documentation is true, however docstrings are _customer facing_ documentation which are python objects whereas code comments are internally facing (and best practice is that comments are not python objects such as string) - hence the importance of being able to distinguish them from each other.

I share the opinion of others in this thread that this PR should be reverted. I also think that this PR does not respect what is most significant for the python color themes in VSCode: distinguishing the different features of the python language to the developer.

**Comment by Afrozodiacc** [comment]
> Forest: Docstrings are documentation. Comments are also documentation. (I actually think we agree here, too.)

Testing the limits of this statement shows why this argument doesn't hold water for formatting things which are of documentation in nature as the same colour...

Trivial code:
```python
class IceCream:
    """Yummy ice cream!"""
    def __init__(self, flavour):
        # Ice cream is typically served cold
        self.temperature_deg_c = -15
        # Support any flavour
        self.flavour = flavour

class IceCreamMachine(Machine):
    """An ice cream machine which dispenses delicious ice cream!"""
    def dispense(n, flavour):
        """Dispenses n number of icecreams in the requested flavour.

        Parameters
        ----------
        n: int
            Number of ice creams to dispense.
        flavour: str
            Any flavour you like so long as it is string.

        Returns
        -------
        list of IceCream
            IceCream class instances
        """
        # Catch any non-str flavour
        if flavour is not str:
            raise TypeError("Flavour must be str.")
        # Create n instances of requested flavour
        icecreams = [IceCream(flavour) for i in range(n)]
        return icecreams
```

Now testing the upper limit of how we can "comment" our code:
```python
class IceCream:
    """Yummy ice cream!"""
    def __init__(self, flavour):
        comment = "Ice_cream_is_typically_served_cold"
        self.temperature_deg_c = -15
        Support_any_flavour = "comment"
        self.flavour = flavour

class IceCreamMachine(Machine):
    """An ice cream machine which dispenses delicious ice cream!"""
    def dispense(n, flavour):
        """Dispenses n number of icecreams in the requested flavour.

        Parameters
        ----------
        n: int
            Number of ice creams to dispense.
        flavour: str
            Any flavour you like so long as it is string.

        Returns
        -------
        list of IceCream
            IceCream class instances
        """
        Catch_any_non_str_flavour = "comment"
        if flavour is not str:
            raise TypeError("Flavour must be str.")
        "Create n instances of requested flavour"
        icecreams = [IceCream(flavour) for i in range(n)]
        return icecreams
```
Comment: Ok now we are using variables to comment our code, it's not python best practice but it is effectively documenting what the code is doing. Those variable assignments which are documentation should now be color formatted the same as code comments. Using variable assignments this way creates an object as documentation much the same as using unassigned str to document does.

Now testing the lower limit of how we can document our code:
```python
class IceCream:
    def __init__(self, flavour):
        self.temperature_deg_c = -15
        self.flavour = flavour

class IceCreamMachine(Machine):
    def dispense(n, flavour):
        if flavour is not str:
            raise TypeError("Flavour must be str.")
        icecreams = [IceCream(flavour) for i in range(n)]
        return icecreams
```
Comment: In this case without the use of docstrings, unassigned str, comments or "comment" variable assignments we have sufficiently (though minimally) documented code through careful selection of class, attribute, method names and variable assignments. Python code is often written in ways where it ends up documenting itself. Should this also be formatted the same colour as code comments?

**Comment by Afrozodiacc** [comment]
> > Forest: Docstrings are documentation. Comments are also documentation. (I actually think we agree here, too.)
> 
> Testing the limits of this statement shows why this argument doesn't hold water for formatting things which are of documentation in nature as the same colour...

To summarise this demonstration shows that:
- Docstrings are documentation
- Comments are typically used as documentation
    - Note: sometimes they are used to prevent lines of code from being executed and not for documentation.
- Variables and unassigned str can be used as documentation
- Code itself can be used documentation

Hence, we should not colour docstring/comment/variables with our inference of intent (such as documentation) but color them by what they are as features of the python language (otherwise everything will be coloured the same).

**Comment by qiweiche** [comment]
I used docstring to write function docs which will appear in the hint and comments just for comments. as suggested by PEP8: https://peps.python.org/pep-0008/#documentation-strings

This change is messing up readability of my code. And previous style has been there for several years, if this is the new standard we are pushing, at least we should provide backwards compatibility?

If you would say you don't like the color in your theme, then we have lots of users like this orange color in Dark+

Please revert and provide better solution.

**Comment by minhaics** [comment]
Dear @ElectricRCAircraftGuy, could you please revert back to the old color scheme where docstrings had a distinct color for easier readability. This is beneficial and user-friendly for programmers' eyes. It's hard to believe that such a terrible request was approved, causing this disruption.

**Comment by ElectricRCAircraftGuy** [comment]
@minhaics , 

Reminder, here is how to fix the colors for yourself:
1. [Restarted Visual Studio Code, and now Python docstrings are colored like comments. How do I return it to the string color?](https://stackoverflow.com/q/76436694/4561887)
    1. [Answer by @starball5](https://stackoverflow.com/a/76436807/4561887)
    2. My own answer: [How to colorize Python docstrings to your liking; ex: like strings](https://stackoverflow.com/a/76440715/4561887)

If you do this change and then click the little person icon in the bottom-left of VSCode, and sign in with GitHub (or one of the other sign-in options) and turn on sync, it will apply this change to all instances of VSCode you have, on all computers you have.

**Comment by minhaics** [comment]
hi @ElectricRCAircraftGuy 
I think that having the color of the docstring the same as the comment in green is your personal preference. Personally, I find green harder to read and distinguish than orange. Moreover, programmers often pay more attention to and read docstrings more than comments. Therefore, it is not reasonable to have the docstring in the same green color as the comment. You should restore it to what it was before. If you like, you can customize it in the settings section. It's not appropriate to make everyone follow your preference. Since you introduced the setting, there have been many objections, as it is not reasonable to require everyone to go into settings to make adjustments. I sincerely hope that you will restore it to what it used to be!

**Comment by minhaics** [comment]
dear @ElectricRCAircraftGuy
Python itself is unlike other languages; it is a language that doesn't use curly braces {}. Therefore, it can be quite challenging to identify the beginning of a function based solely on the "def func" declaration. This is insufficient for the programmer's eyes. That's where the presence of a docstring comes to the rescue. I find it very helpful to easily identify the existence of a function by adding a docstring to it. Hence, I frequently look at the docstring. A docstring with beautiful colors and different from the ugly default color is preferable.

**Comment by ElectricRCAircraftGuy** [comment]
> but I tried to find the specific code for it and couldn't. I don't know what it was

@minhaics , what color scheme do you have? I'll give you the code. 

The color you had is the color of a string. Open a Python file and type `s = "string"`, then use the developer inspector, as I explain, to inspect that string and get its color code.

**Comment by scholer** [comment]
This PR is based on the wrong assumption that in Python, tripple-quote strings that are not assigned to a variable are "block comments". This is not correct. It is just a string that is not assigned. Python doesn't have any true options for multi-line comments, like `/*` ... `*/` in C. 

True, some people highjack tripple-quotes to quickly comment out blocks of code, but that is super flaky. For instance if the code you want to comment out already contains tripple-quoted strings. Or if used inside brackets (where python will concatenate strings if they are only separated by newlines and whitespace). The proper way to comment out parts of code in Python is to highlight the lines you want to comment out and ask your IDE to comment the selected lines (ctrl + / in vscode).

The `"""SAMPLE OUTPUT"""` block-comment example in the original post could be just as easily be written using normal `# SAMPLE OUTPUT:` So, if you "don't like the yellow color" that your IDE is using for strings to apply to your tripple-quoted "comments", just do that—use `#` for those comments. In fact, you could argue that retaining the original behavior is also favorable even in these instances, since it gives people the option to *highlight* certain "comments". So, if I wanted to actually highlight the "SAMPLE OUTPUT" comment, I can either use a tripple-quoted string and make it stand out, or I can use `#` to make it a bit more muted.

Other questions:
- Do you also want to color other expressions that are not assigned to a variable as comments? Like `2 + 2`? 
- If unassigned tripple-quoted strings are treated as block comments, what about tripple-quoted f-strings? Why should `"""HELLO"""` be colored as a comment, but an unassigned `f""""HELLO""""` be colored as a string? (The answer is obvious, since an unassigned f-string can invoke functions. But symmetry says that they should appear similarly in the editor.)
- What about "notebook-like" files, where many notebook applications (e.g. Jupyter) will output the last statement in a cell, if it is not assigned to a variable, including tripple-quoted strings. Since some notebook applications are just using plain .py files as notebooks (e.g. Databricks), the editor can't really know which tripple-quoted strings are at the end of a cells and which are not.

All in all, I understand the rationale for this PR, but it just seems like it can have too many unexpected/unintended side-effects or edge-cases. Which is what can easily happen when you change the interpretation of basic parts of a language.

Finally, as everyone else have already pointed out: *Docstrings*, in particular, are certainly not comments, since they affect the output being produced by the interpreter. They should probably have their own color-theme option in the editor. (And if you want your docstrings to be colored the same color as comments, you can pick a theme that does that.)

**Comment by minhaics** [comment]
Hi @ElectricRCAircraftGuy 
Please restore it as it originally belonged before, so users don't have to manually customize it in the settings in the .json file

**Comment by minhaics** [comment]
Hi @ElectricRCAircraftGuy
You said, "The color you had is the color of a string. Open a Python file and type s = "string", then use the developer inspector, as I explain, to inspect that string and get its color code. If you give me the name of your color scheme, I'll give it to you." These manual steps are inconvenient for users, even in VS Code, I don't have a default tool to find the color code of the displayed string. I rely on and trust the VS Code core team for what they have done for Python. I prefer the default settings without additional customization. However, you have ruined it, so once again, please restore it to its original state, as it used to be. Thanks!
